src/Workspaces/Core/Portable/EmbeddedLanguages/AbstractEmbeddedLanguageFeatureService.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.EmbeddedLanguages;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages
+{
+    internal abstract class AbstractEmbeddedLanguageFeatureService<TService>
+        where TService : IEmbeddedLanguageFeatureService
```

Comment:
this was extracted from AbstractEmbeddedLanguageClassificationService  so it could be shared by brace matching as well so we don't need to duplicate.